caf: ble_bond: Introduce CAF BLE bond implementation (#6525)
nrfconnect/sdk-nrf: applications/machine_learning/README.rst

The application supports a button that is used to switch between data forwarding and running the machine learning model.
You can trigger the change by pressing and holding the button for longer than 5 seconds.

.. note::

>    If a given configuration supports Bluetooth, then pressing a button shortly after application boot triggers removing Bluetooth bond.

Reviewer: The note is not clear that which button and what exactly happens when the button is pressed.

Author: Refered to the button as `the button`. There is only one button used to control app state. Is it ok now ?

>    If a given configuration supports Bluetooth, then pressing the button shortly after application boot triggers removing Bluetooth bond.

Reviewer: ```suggestion
   If a given configuration supports Bluetooth, the Bluetooth bond can be removed by pressing a button shortly after the application has booted.
```
I think we should mention which button will remove the bond (like it a reset button or something else)

Author: The button used to switch between data forwarding and running the machine learning model. It's actually configurable and it may depend on used board.

>    If a given configuration supports Bluetooth, then two sequences of the button press and release in a short time interval, shortly after application boot, trigger removing Bluetooth bond.

Reviewer: Rephrased a bit.

>    If a given configuration supports Bluetooth, then two sequences of the button press and release in a short time interval, shortly after application boot, trigger removing Bluetooth bond.

Author: Fixed

>    If a given configuration supports Bluetooth, then two sequences of the button press and release in a short time interval occurring shortly after application boot trigger removing Bluetooth bond.

Reviewer: ```suggestion
   If a given configuration supports Bluetooth, then button press and release in a short time interval, shortly after the application boot trigger, removes the Bluetooth bond.
```

Author: This is not true now. We need to perform double click (`two sequences of the button press and release in a short time interval`)

Author: I meant that the 2 sequences should appear shortly after application is booted.
